Tests/TuistGeneratorTests/Generator/Mocks/MockWorkspaceGenerator.swift

```diff
@@ -3,17 +3,24 @@ import Foundation
 import TuistCore
 import TuistCoreTesting
 import TuistSupport
+import XcodeProj
 @testable import TuistGenerator
 
 final class MockWorkspaceGenerator: WorkspaceGenerating {
     var generateWorkspaces: [Workspace] = []
-    var generateStub: ((Workspace, AbsolutePath, Graphing, TuistConfig) throws -> AbsolutePath)?
+    var generateStub: ((Workspace, AbsolutePath, Graphing) throws -> WorkspaceDescriptor)?
 
-    func generate(workspace: Workspace,
-                  path: AbsolutePath,
-                  graph: Graphing,
-                  tuistConfig: TuistConfig) throws -> AbsolutePath {
+    func generate(workspace: Workspace, path: AbsolutePath, graph: Graphing) throws -> WorkspaceDescriptor {
         generateWorkspaces.append(workspace)
-        return (try generateStub?(workspace, path, graph, tuistConfig)) ?? AbsolutePath("/test")
+        return try generateStub?(workspace, path, graph) ?? stub
```

Comment:
Considering that the definition of this method throws, I'd throw an error if `generateStub` is nil. That way, if the tests is not well implemented and they forget to set the stub, they get an error and prevent them from writing a test that relies in a internal descriptor that might change in the future.